pkg/repo/chartrepo.go

```diff
@@ -110,15 +112,14 @@ func (r *ChartRepository) Load() error {
 //
 // cachePath is prepended to any index that does not have an absolute path. This
 // is for pre-2.2.0 repo files.
-func (r *ChartRepository) DownloadIndexFile(cachePath string) error {
-	var indexURL string
+func (r *ChartRepository) DownloadIndexFile(cachePath string, fs afero.Fs) error {
```

Comment:
This changes the public function signature in a non-backwards compatible manner. A breaking change, like this, would need to be on the v3 branch or the change would need to happen in a manner than doesn't break function backwards compatibility.

Reply:
@mattfarina Ah. I overlooked that! It's a breaking change, yes. What do you think about having fs a field in ChartRepository? Like the `Client` Would it make sense? And I'll also test it for backwards compatibility by making sure that if fs is not present, it will be initialized. 

And the reason for putting fs is - for dependency injection to make testing easier. This way, it's possible to inject a in-memory fs for testing and os fs for actual execution

Reply:
@mattfarina I have made changes in such a way that it doesn't break the API and is backwards compatible. Could you check now?